packages/kbn-config-schema/src/types/union_type.test.ts

```diff
@@ -62,22 +62,42 @@ test('handles object', () => {
 });
 
 test('handles object with wrong type', () => {
-  const type = schema.oneOf([schema.object({ age: schema.number() })]);
+  const type = schema.oneOf([schema.object({ age: schema.number() }), schema.string()]);
 
   expect(() => type.validate({ age: 'foo' })).toThrowErrorMatchingInlineSnapshot(`
-"types that failed validation:
-- [0.age]: expected value of type [number] but got [string]"
-`);
+    "types that failed validation:
+    - [0.age]: expected value of type [number] but got [string]
+    - [1]: expected value of type [string] but got [Object]"
+  `);
 });
 
-test('includes namespace in failure', () => {
+test('use shorter error messages when defining only one type', () => {
   const type = schema.oneOf([schema.object({ age: schema.number() })]);
 
+  expect(() => type.validate({ age: 'foo' })).toThrowErrorMatchingInlineSnapshot(
+    `"[age]: expected value of type [number] but got [string]"`
+  );
+});
```

Comment:
`joi.alternatives` is now doing an optimization when only one alternative is registered to just delegate all the calls to the single alternative. This is why I had to adapt some tests to have multiple subtypes.

Reply:
> joi.alternatives is now doing an optimization when only one alternative is registered

Is there a valid use for a single option in `oneOf`? Maybe we should require at least 2 options to be declared?

Reply:
> Is there a valid use for a single option in oneOf

I can imagine use cases such as generic builders

```ts
const createAllowedValueSchema = (values: string[]) => 
    schema.oneOf([...values.map((value) => schema.litteral(value))])
```

where it would be awkward for consumers to check if the length of `values` is one to use a different definition instead (I saw a few in the security solution schema code, and we may have some other elsewhere)